plugins/org.jboss.reddeer.eclipse/src/org/jboss/reddeer/eclipse/m2e/core/ui/wizard/MavenImportWizard.java

```diff
@@ -0,0 +1,148 @@
+package org.jboss.reddeer.eclipse.m2e.core.ui.wizard;
+
+import org.jboss.reddeer.common.condition.AbstractWaitCondition;
+import org.jboss.reddeer.common.wait.AbstractWait;
+import org.jboss.reddeer.common.wait.TimePeriod;
+import org.jboss.reddeer.common.wait.WaitUntil;
+import org.jboss.reddeer.common.wait.WaitWhile;
+import org.jboss.reddeer.core.condition.JobIsRunning;
+import org.jboss.reddeer.core.condition.ShellWithTextIsActive;
+import org.jboss.reddeer.core.condition.ShellWithTextIsAvailable;
+import org.jboss.reddeer.core.exception.CoreLayerException;
+import org.jboss.reddeer.jface.wizard.ImportWizardDialog;
+import org.jboss.reddeer.swt.exception.SWTLayerException;
+import org.jboss.reddeer.swt.impl.button.PushButton;
+import org.jboss.reddeer.swt.impl.combo.LabeledCombo;
+import org.jboss.reddeer.swt.impl.shell.DefaultShell;
+
+/**
+ * Wizard for importing an existing maven project.
+ * 
+ * @author apodhrad
+ *
+ */
+public class MavenImportWizard extends ImportWizardDialog {
+
+	public static final String TITLE = "Import Maven Projects";
+
+	/**
+	 * Constructs ImportWizard with "Maven > Existing Maven Projects".
+	 */
+	public MavenImportWizard() {
+		super("Maven", "Existing Maven Projects");
+	}
+
+	/**
+	 * Activate the wizard.
+	 */
+	public void activate() {
+		new DefaultShell(TITLE);
+	}
+
+	/**
+	 * Sets a root directory.
+	 * 
+	 * @param path
+	 *            Path
+	 */
+	public void setRootDirectory(String path) {
+		activate();
+		new LabeledCombo("Root Directory:").setText(path);
+	}
+
+	/**
+	 * Click on 'Refresh' button.
+	 */
+	public void refresh() {
+		activate();
+		new PushButton("Refresh").click();
+	}
+
+	/**
+	 * Waits until a project is loaded (usually after refresh).
+	 * 
+	 * @param timeout
+	 *            Timeout
+	 */
+	public void waitUntilProjectIsLoaded(TimePeriod timeout) {
+		activate();
+		new WaitUntil(new ProjectsIsLoaded(), timeout);
+	}
+
+	@Override
+	public void finish() {
+		finish(TimePeriod.VERY_LONG);
+	}
+
+	@Override
+	public void finish(TimePeriod timeout) {
+		new PushButton("Finish").click();
+		new WaitWhile(new ShellWithTextIsActive(TITLE), TimePeriod.NORMAL, false);
+		try {
+			activate();
```

Comment:
What happens/may happen when you click on finish:
- Import Maven Projects wizard is closed
MAY HAPPEN: another Import Maven Projects wizard is opened (different shell, same name..ouch) which informs about some missing maven plugins (typically lifecycle). This wizard contains "Resolve All Later" and "Finish" button. 

Your solution to this situation is not optimal - first you wait for ShellWithTextIsActive (generally not a good idea - some window managers dont report correctly what is active and what is not) then you try to activate some "random"  shell and push buttons. I propose to implement matcher which will look for shell with these buttons and avoid using ShellWithTextIsActive.